lib/services/services.go

```diff
@@ -52,9 +55,39 @@ func (s *ServiceRegistry) StartAll() {
 	s.logger.Debug("All services started.")
 }
 
+// PauseServices pauses key services before shutdown to allow a graceful shutdown
+func (s *ServiceRegistry) PauseServices() {
+	s.logger.Infof("Pausing key services")
+	// Pause the sync and state service to allow for graceful shutdown
+	syncService, ok := s.services[(reflect.TypeOf(&sync.Service{}))].(*sync.Service)
+	if ok && syncService != nil {
+		err := syncService.Pause()
+		if err != nil {
+			s.logger.Errorf("Error pausing sync service: %s", err)
+		}
+	} else {
+		s.logger.Errorf("Error getting sync service")
+	}
+
+	stateService, ok := s.services[(reflect.TypeOf(&state.Service{}))].(*state.Service)
+	if ok && stateService != nil {
+		err := stateService.Pause()
+		if err != nil {
+			s.logger.Errorf("Error pausing state service: %s", err)
+		}
+	} else {
+		s.logger.Errorf("Error getting state service")
+	}
+
+	s.logger.Infof("Paused key services")
+}
+
 // StopAll calls `Service.Stop()` for all registered services
 func (s *ServiceRegistry) StopAll() {
 	s.logger.Infof("Stopping services: %v", s.serviceTypes)
+
+	s.PauseServices()
+
```

Comment:
Looks a little bit odd to me that we have a method called `PauseServices` that pauses only two specific services (sync and block state) and then we have a loop that goes over the services types (that includes sync and block state) and stop them again.

I think this is convoluted and one option would be to have some list that holds the order to stop services, then you can always ensure that sync and state services are going to be stopped first then any other service

Reply:
Agree, we discussed it with @jimjbrettj in a private chat last week, the best solution should be having a "dependency list / tree" to start and stop the services following that specific order. But I think this PR aims to be a hotfix instead of a final solution

Reply:
I believe that this modification can be included in this PR

Reply:
updated!